src/rendering/mod.rs

```diff
@@ -107,6 +107,22 @@ impl OrbitCamera {
         let world_space_delta = transform * delta * 0.1;
         self.target -= world_space_delta;
     }
+
+    pub fn set_target(&mut self, target: nalgebra::Point3<f32>) {
+        self.target = target;
+    }
+
+    pub fn set_range(&mut self, range: f32) {
+        self.range = range;
+    }
+
+    pub fn set_azimuth(&mut self, azimuth: f32) {
+        self.azimuth = azimuth;
+    }
+
+    pub fn set_elevation(&mut self, elevation: f32) {
+        self.elevation = elevation;
+    }
```

Comment:
since you made the attributes public you don't really need these right?
Either keep them private and provide the getters you need for the float conversion, or remove these public functions.

Unless there is some rust reason to have these.